src/js/client.ts

```diff
@@ -144,4 +158,23 @@ export class ReactNativeClient extends BaseClient<ReactNativeClientOptions> {
       );
     }
   }
+
+  /**
+   *
+   */
+  private _attachClientReportTo(envelope: ClientReportEnvelope): void {
+    const outcomes = this._clearOutcomes();
+
+    if (outcomes.length > 0) {
+      const clientReportItem: ClientReportItem = [
+        { type: 'client_report' },
+        {
+          timestamp: dateTimestampInSeconds(),
+          discarded_events: outcomes,
+        },
+      ];
+
+      envelope[items].push(clientReportItem);
```

Comment:
What if the envelope already has a `clientReportItem` item? is it 2? should we merge them?

Reply:
It can happen, but if the server can handle it, why should we? 

Maybe to save a bit of space? Would that be a reason?

Reply:
It's cheaper to process on the server, has fewer data to transfer over the wire, and space in the disk, I recall folks from the ingest team also preferring a single client report per envelope whenever possible.

Reply:
Okay that's what I thought.